x/stakeibc/keeper/validator_selection.go

```diff
@@ -244,26 +258,33 @@ func (k Keeper) GetValidatorDelegationDifferences(ctx sdk.Context, hostZone type
 	return delegationDeltas, nil
 }
 
-// This will get the target validator delegation for the given hostZone
-// such that the total validator delegation is equal to the finalDelegation
-// output key is ADDRESS not NAME
-func (k Keeper) GetTargetValAmtsForHostZone(ctx sdk.Context, hostZone types.HostZone, finalDelegation sdkmath.Int) (map[string]sdkmath.Int, error) {
+// This will split a total delegation amount across validators, according to weights
+// It returns a map of each portion, key'd on validator address
+// Validator's with a slash query in progress are excluded
+func (k Keeper) GetTargetValAmtsForHostZone(ctx sdk.Context, hostZone types.HostZone, totalDelegation sdkmath.Int) (map[string]sdkmath.Int, error) {
 	// Confirm the expected delegation amount is greater than 0
-	if !finalDelegation.IsPositive() {
+	if !totalDelegation.IsPositive() {
 		return nil, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest,
-			"Cannot calculate target delegation if final amount is less than or equal to zero (%v)", finalDelegation)
+			"Cannot calculate target delegation if final amount is less than or equal to zero (%v)", totalDelegation)
+	}
+
+	// Ignore any validators with a slash query in progress
+	validators := []types.Validator{}
+	for _, validator := range hostZone.Validators {
+		if !validator.SlashQueryInProgress {
+			validators = append(validators, *validator)
+		}
 	}
 
 	// Sum the total weight across all validators
-	totalWeight := k.GetTotalValidatorWeight(hostZone)
+	totalWeight := k.GetTotalValidatorWeight(validators)
 	if totalWeight == 0 {
 		return nil, errorsmod.Wrapf(types.ErrNoValidatorWeights,
 			"No non-zero validators found for host zone %s", hostZone.ChainId)
 	}
 	k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId, "Total Validator Weight: %d", totalWeight))
 
-	// sort validators by weight ascending, this is inplace sorting!
-	validators := hostZone.Validators
+	// sort validators by weight ascending
 	sort.SliceStable(validators, func(i, j int) bool { // Do not use `Slice` here, it is stochastic
```

Comment:
To avoid any uncertainty about the effect of no longer sorting the validators, why not move this sorting block up, above the block that filters out validators who have a slash query in progress? That way the vals on the HZ get sorted here too, and the filtering still applies on the copy of the val array

Reply:
I think we should remove the sorting unless we can think of a specific reason we'd need it. It's a dangerous operation (remember we once had a chain halt since it was nondeterministic). Informal also specially recommended we remove it for the case of unbondings.

Reply:
I can't think of why we need it, but it _is_ currently being sorted on mainnet with no issues. So it feels to me that removing the sort could be the change that causes nondeterminism, rather than keeping it. 

The scenario we def want to avoid is re-ordering the vals and saving them back to the HZ without a deterministic sort. Don't think that's happening here even if we remove the sort, so we should be safe. 

Agree we should get to the bottom of why we originally sorted inplace... @asalzmann @shellvish do you recall why we sorted inplace here?

Reply:
I don't think the inplace sort was intentional. I think it was a consequence of us calling it off the host zone struct which has a list of pointers

Reply:
Looking through this again, I don't _think_ we need to sort at all. The only impact I can think of is the validator that gets the "overflow" will change, although I don't think this has a meaningful affect in practice.

The reason I think it shouldn't matter is that the output of this function is a map, which isn't ordered to begin with. So, as long as the contents of the map are the same, we shouldn't need to care about the order in which they were generated. 

The one thing that makes me nervous though is that we specifically are doing a reverse sort. I'm not sure why we chose to do that, which makes me think there might have been a reason for it. This was also done around launch though so it's very possible it was just an extra precaution because it makes the function easier to reason about.

@asalzmann might have a better memory of why we chose to sort!

Reply:
I think the non-deterministic part here was the sorting function itself. I think we used to use `slice.Sort` (which turned out to be non-deterministic) and then it was switched to a combo of `slice.SortStable` + reversing the list first (which is deterministic).

Also, worth noting, this has since been updated so that the validator name is [used as a tie breaker ](https://github.com/Stride-Labs/stride/blob/670369ef5a3af070da94ddbd7ed5b0dd832b7f7c/x/stakeibc/keeper/validator_selection.go#L287-L294)when two validator's have the same weight. So *technically*, we don't even need to use SortStable anymore (although we should still just to be safe)

Reply:
@shellvish considering the new tie breaker - do you still think the overflow validator changes? I don't think it does

Reply:
I think that's right re:tiebreaker, but I think it does change the overflow validator (because, I believe, we don't sort the validators on the Host Zone struct). Regardless, I do think we can remove the sorting entirely given the map struct, but would be good to understand why we sorted in the first place!

Reply:
leaving removed